boards/sim/sim/sim/scripts/Make.defs

```diff
@@ -100,6 +100,24 @@ CXXFLAGS := $(ARCHWARNINGSXX) $(ARCHOPTIMIZATION) \
 CPPFLAGS := $(ARCHINCLUDES) $(ARCHDEFINES) $(EXTRAFLAGS)
 AFLAGS := $(CFLAGS) -D__ASSEMBLY__
 
+ifeq ($(CONFIG_LIBCXX),y)
+  CXXFLAGS += -std=c++14 -D_LIBCPP_STD_VER=14
```

Comment:
why add this or not c++17? libc++ can work with c++11 too.

Reply:
it was the version what i happened to be using and tested.
as this version of the pr (after the rebase) is not tested and i mostly lost my motivation to test it on linux, i guess i will just remove this.